realsense2_camera/src/realsense_node.cpp

```diff
@@ -93,25 +106,154 @@ BaseRealSenseNode::BaseRealSenseNode(ros::NodeHandle& nodeHandle,
     filters[3].is_enabled = false;
 
     _prev_camera_time_stamp = 0;
+
+    publishTopics();
 }
 
-void BaseRealSenseNode::publishTopics()
+void RealSenseNode::createParamsManager() {
+    auto pid_str = _dev.get_info(RS2_CAMERA_INFO_PRODUCT_ID);
+    uint16_t pid;
+                std::stringstream ss;
+                ss << std::hex << pid_str;
+                ss >> pid;
+                ROS_WARN_STREAM ("pid " << pid);
+
+                switch(pid)
+                {
+                case SR300_PID:
+                    _params = std::unique_ptr<SR300ParamManager>(new SR300ParamManager);
+                    break;
+                case RS400_PID:
+                   _params = std::unique_ptr<D400ParamManager>(new D400ParamManager);
+                    break;
+                case RS405_PID:
+                    _params = std::unique_ptr<D400ParamManager>(new D400ParamManager);
+                    break;
+                case RS410_PID:
```

Comment:
Why are only RS410_PID and RS460_PID grouped together? Most of this list can be grouped together since most of it just calls `_params = std::unique_ptr<D400ParamManager>(new D400ParamManager);`